code/game/atoms.dm

```diff
@@ -280,22 +280,22 @@ var/list/blood_splatter_icons = list()
 
 //returns 1 if made bloody, returns 0 otherwise
 /atom/proc/add_blood(mob/living/carbon/M)
+	if(!M || !M.has_dna() || rejects_blood())
+		return 0

```

Comment:
I don't like these checks, isn't this proc runtiming because of changelings fuck ups with dna?